test/java/seedu/addressbook/parser/ParserTest.java

```diff
@@ -304,4 +312,18 @@ private void parseAndAssertIncorrectWithMessage(String feedbackMessage, String..
         assertTrue(result.getClass().isAssignableFrom(expectedCommandClass));
         return (T) result;
     }
+    
+    /**
+     * Returns a concatenated version of the printable strings of each object.
+     */
+    String getPrintableString(Printable... printables) {
+        String output = "";
+        for (Printable p : printables) {
+            if (output.isEmpty() == false) {
+                output += NEWLINE;
```

Comment:
Doesn't this result in an extra line break at the end? You could have used `StringJoiner` or `String.join` instead.